Update nl.js

milovanderlinden opened the pull request.

milovanderlinden committed:
Update nl.js

patcon commented:
Thanks! LGTM

Quick Q: Are there variants of dutch we should accommodate here, or does this feel like a universal improvement?

metasoarous commented:
Hi @milovanderlinden. Thanks for submitting this!

From translate.google.com it looks like the origin "ozeke" is probably supposed to be "ozeker", translating to "unsure". Meanwhile "Geen mening" comes out closer to "no opinion". I'm inclined to think "ozeker" is closer to what we want, since you might have an opinion, but feel conflicted. In fact, when we explain the tool we typically say "participants can vote agree, disagree, or pass if unsure". So I'm inclined to suggest we fix "ozeke" -> "ozeker".

Regarding "resterend", it seems that may be a somewhat more formal alternative to "overblijvend", so I'm happy to take that suggestion.

https://hinative.com/en-US/questions/4831483

Thanks again!

metasoarous commented:
@patcon I'm not sure why the linter dislikes that bit of JSX I pushed up the other day. It seems to think it should be one line instead of many, but it's failing to take into account that the other surrounding blocks are a bit longer, and this one would look at of place on one line.

Do you know how to soften this up a bit?

Thanks!

metasoarous committed:
Replace "Geen mening" with "Onzeker"

"Onzeker" means "unsure", while "Geen Mening" seems to mean something closer to "No opinion". I think "Onzeker" is closer to what we want here.

patcon commented:
I just adopted what prettier thinks in another PR -- is that ok?

I can see why it thinks what it thinks, and exceptions feel more hassle than they're worth. Obvs no objection to your changing it, but I guess I kinda feel the value of a linter is in not second-guessing it too much for edgecases :)

But on the substance of this issue, maybe we could draft a little styleguide of the "intentions" of the language, for when people generously offer translation support. I feel like there are some intuitions the Polis Team has had. (As always, preference is to store this in repo with code, where others can contribute, since it has to do with contribution, but if it must go in knowledgebase, that's cool too)

First thoughts:
- something about "comment" vs "statement" considerations. choosing words that don't hint at "responding to one another" in the given language (like back-and-forth of comments on Reddit on blog comments).
- action verbs: "vote" [agreement/disagreement/pass] + "submit" [comment/statement]
- "pass" being neutral stand-in for situations where [explain contexts we expect ppl to click it, like "poorly worded" or "partially true", "disagree with wording", etc etc]
  - chosen over other, less apt terms like "no opinion" or "skip" or [others that were considered]

metasoarous commented:
To be honest, I'd just as soon nuke the linter entirely. I find they're generally more trouble than they're worth, especially if there's decent test coverage, and doubly especially when the linter gets overzealous. I get that other people like them though, and there's always the chance it'll catch a bug the tests miss, so this is me trying to meet in the middle.

I can look into whether this specific check can be turned off, but haven't messed with JS linters much, so I'd appreciate your eyes on it if this is something you're more familiar with.

> I just adopted what prettier thinks in another PR -- is that ok?

Not seeing another PR for this. Do you mean this historically, as in an already merged PR?

I agree it would be nice to have some content somewhere regarding how to craft good translations, and what the intended/desired language is. Perhaps these could even be comments inline in the English strings file, since I think this is what people typically copy from.

It might also be good to split apart the UI chunks that are actually in use from those that aren't. Some of those are from old bits of UI that aren't any longer in use, and it should be easy for translators to prioritize the strings that _are_ still in use.

Thanks again

patcon commented:
> To be honest, I'd just as soon nuke the linter entirely.

No block on nuking (your call ultimately), but my recollection is that @ballPointPenguin and I were fans (?), as was CS :heart: Suggestion: let's just keep asking new people who show up what they prefer? If newcomers or active contributors don't appreciate it, I want to let it go too :)

> Do you mean this historically, as in an already merged PR?

Yeah, sorry, I mistook this for something self-mergeable, and jumped the gun in resolving it in another PR before we even interacted.

I'll resolve this one. It has ignore :) (prettier's philosophy is that it doesn't really do fine-tuned "rules": https://prettier.io/docs/en/option-philosophy.html)

colinmegill commented:
Can't prettier auto run and fix community PRs?

For frontend, prettier and eslint should definitely remain.

On Tue, Dec 8, 2020, 9:04 PM Patrick Connolly <notifications@github.com>
wrote:

> To be honest, I'd just as soon nuke the linter entirely.
>
> No block on nuking (your call ultimately), but my recollection is that
> @ballPointPenguin <https://github.com/ballPointPenguin> and I were fans
> (?), as was CS ❤️ Suggestion: let's just keep asking new people who show
> up what they prefer? If newcomers or active contributors don't appreciate
> it, I want to let it go too :)
>
> Do you mean this historically, as in an already merged PR?
>
> Yeah, sorry, I mistook this for something self-mergeable, and jumped the
> gun in resolving it in another dependabot PR before we even interacted.
>
> I'll resolve this one. It has ignore (prettier's philosophy is that it
> doesn't really do fine-tuned "rules":
> https://prettier.io/docs/en/option-philosophy.html)
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/pol-is/polis/pull/673#issuecomment-741529307>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQGGOTJ7OZH4KLF2YUVPTST4APBANCNFSM4UQHVFSQ>
> .
>

patcon commented:
It can, with clickable PR suggestions. Tracked +1 over here: https://github.com/pol-is/polis/issues/162#issuecomment-646330324

Convo was more about whether to (a) "correct the linter when it makes weird comments" or (b) "make nice with the linter, it's thanksgiving, just smile and nod" :)

(I am the linter-enabling millennial in this confusing analogy)

Anyhow, all good now, I think! Thanks!

metasoarous commented:
OK; Thanks @patcon. I see your #674 PR now, and have merged. It seems like there's a way to be more specific about which rules you're ignoring in the future, but I was having trouble finding this specific rule.

Would love to get clickable PR suggestions set up if possible.

Thanks again!
